File: Python-class/string_lab.py
def string_analyzer(text):
    vowels= "aeiouAEIOU"
    vowel_count=0

    for char in vowels:
        if char in text:
            vowel_count+=1

    cleaned_text=text.replace(" ","").lower()
    is_palindrome = cleaned_text==cleaned_text[::-1]

    print(f"No. of vowels:{vowel_count}")
    if is_palindrome:
        print("It is a palindrome!")
    else:
        print("Not a palindrome!")
#Main Program
print("---String Manipulation---")
user_input =input("Enter a word or a sentence: ")
string_analyzer(user_input)

File: Java-class/StringLab.java
import java.util.Scanner;

public class StringLab {
    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
        System.out.println("Enter a word or a sentence: ");
        String text = sc.next();

        String vowels = "aeiouAEIOU";
        int vowel_count=0;

        for(int i=0;i<text.length();i++){
            char ch=text.charAt(i);

            if(vowels.indexOf(ch)!= -1){
                vowel_count++;
            }

        }

        Boolean is_palindrome =true;
        int start=0;
        int end=text.length() - 1;

        while(start<end){
            if(Character.toLowerCase(text.charAt(start)) != Character.toLowerCase(text.charAt(end))){
                is_palindrome=false;
                break;
            }
            start++;
            end--;
        }

        System.out.println("No. of vowels: " + (vowel_count));
        if(is_palindrome){
            System.out.println("It is a Palindrome!");
        }
        else{
            System.out.println("It is not a Palindrome!");
        }
        sc.close();
    }
}
